pr_watch/github.py

```diff
@@ -183,6 +192,19 @@ def parse_date(date):
     return datetime.strptime(date, '%Y-%m-%dT%H:%M:%SZ')
 
 
+def ensure_rate_limit(url):
+    """
+    Check the ratelimit to ensure that we'll be able to successfully make a request.
+    """
+    query_type = 'search' if 'search' in url.split('/') else 'core'
+    response = get_object_from_url('https://api.github.com/rate_limit', rate_limited=False)
+    resource = response['resources'].get(query_type, {})
+    if not resource.get('remaining'):
+        warn_string = 'Rate limit exceeded when requesting a \'{}\' resource at {}'.format(query_type, url)
+        logger.warning(warn_string)
+        raise RateLimitExceeded(warn_string)
```

Comment:
I don't see how this check is useful.  We basically just replace one traceback in the logs by a different one, and instead of trying to make the actual request we make an API request to query the rate limit, so we don't gain anything.  We actually increase the total number of API requests, and we have additional code that can break, so I think we should just try the request and let Github check the rate limit for us.

Reply:
@smarnach, I think you're right. At first I was concerned, because GitHub actually just returns a 403 response code when we hit the limit, rather than a much-more-helpful 429, but we can check a header to see if we're out of requests, and use that to make sure we behave properly.